src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -178,10 +179,15 @@ def _resolve_single_import(owners, import_name) -> ImportResolveResult:
         else:
             return ImportResolveResult(ImportOwnerStatus.unowned)
 
-    return {
-        imp: _resolve_single_import(owners, imp)
-        for owners, (imp, inf) in zip(owners_per_import, parsed_imports.items())
-    }
+    result = {}
+    for owners, (imp, inf) in zip(owners_per_import, parsed_imports.items()):
+        resolved_single_import = _resolve_single_import(owners, imp)
+        if ignore_self_imports:
+            if not resolved_single_import.address or (resolved_single_import.address[0] != address):
```

Comment:
`resolved_single_import.address` is either empty or a singleton tuple, so getting the first and only element.